File: Lab1/program/function.cpp
#include "function.hpp"
#include <iostream>
#include <fstream>
#include <string>

using namespace std;

string input(){
    string name;
    string line;
    string text="";
    bool isFirst = true;
    
//  вводимо імʼя файлу
    cout<<"Enter input file: ";
    cin>>name;
    name = name + ".txt";
    ofstream out(name);
    
    cout<<"\nEnter your text\nPress Enter to go to the next line\nPress Ctrl + D + Enter - to end writing:\n";
//  вводимо текст до натиснення комбінації клавіш
    string ascii_code = "^D";
    getline(cin, line);
    while(line.find(ascii_code)==string::npos){
            getline(cin, line);
            if (isFirst) {
                isFirst = false;
            }
            else {
                text +='\n';//додаємо перехід на наступний рядок перед усіма непершими рядками
            }
            text += line;
        }
    text = text.substr(0, text.size()-2);
    out<<text;
    out.close();
    return name;
}

void output(string name){//виводимо текст файлу
    ifstream in(name);
    string s;
    cout<<"File "<<name<<": \n";
    while(!in.eof()){
        getline(in,s);
        cout<<s<<endl;
    }
    in.close();
}
string changefile(string name){
    string s, i;
    string sep=" ";
    size_t n;
    ifstream in(name);
    string name2;
    cout<<"Enter output file: "; cin>>name2;//називаємо другий файл
    name2+=".txt";
    ofstream out(name2);
    bool isFirst = true;
    while(!in.eof()){
        getline(in, s);
        if (s == "") {
            continue;
        }
        while(1){//знаходимо перуш літеру останнього слова кожного рядка
            n=s.rfind(sep);
            if(n+1 != string::npos){
                i=s.substr(n+1, 1);
                break;
            }
            else{
                s.erase(n,1);
            }
        }
        if (isFirst) {
            isFirst = false;
        } else {
            out << endl;
        }
        out<<i<<s;//додаємо літеру на початок рядка
    }
    out.close();
    in.close();
    return name2;
}

File: Lab1/program/function.py
def capture_text():
    """
    Читаємо текст строками, до натиснення комбінації Ctrl + D/Command + D
    """
    print('Enter your text\nPress Enter to go to the next line\nPress Ctrl + D/Command + D - to end writing')

    import sys
    text = sys.stdin.read()

    # видаляємо зайвий символ \n
    text = text[:len(text) - 1]

    print(f"Full text:\n{text}\n")

    return text


def write_text(name_f: str, text: str):
    """
    Записуємо текст у файл
    """
    file = open(name_f, 'w')
    file.write(text)
    file.close()


def last_word_letter(text: str):
    """
    Вставляємо на початок рядка першу літеру останнього слова у кожному рядку
    """
    lines = text.split('\n')
    for i in range(len(lines)):
        word = lines[i].split()
        letter = word[-1][0]
        lines[i] = letter + lines[i]
    return lines



def write_to_file(name: str, lines: str):
    """
    Записує текст у файл
    """
    write_file = open(name, 'w')
    for i in range(len(lines)):
        write_file.write(f"{lines[i]}\n")

    write_file.close()


def read_file(file_name: str):
    """
    Читає текст з файлу
    """
    file = open(file_name, 'r')
    text = file.read()
    file.close()

    return text